http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -0,0 +1,316 @@
+import io.chrisdavenport.log4cats.StructuredLogger
+import org.broadinstitute.dsde.workbench.google2.util.RetryPredicates
+import org.broadinstitute.dsde.workbench.google2.{GcsBlobName, GoogleStorageService}
+import org.broadinstitute.dsde.workbench.leonardo.RuntimeImageType.{Jupyter, Welder}
+import org.broadinstitute.dsde.workbench.leonardo.config.{AutoFreezeConfig, DataprocConfig, GceConfig, ImageConfig}
+import org.broadinstitute.dsde.workbench.leonardo.dao.DockerDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.{clusterQuery, DbReference, SaveCluster}
+import org.broadinstitute.dsde.workbench.leonardo.http.api.{CreateRuntime2Request, RuntimeServiceContext}
+import org.broadinstitute.dsde.workbench.leonardo.http.service.LeonardoService._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.RuntimeServiceInterp._
+import org.broadinstitute.dsde.workbench.leonardo.model._
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage.CreateCluster
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.{google, TraceId, UserInfo, WorkbenchEmail}
+
+import scala.concurrent.ExecutionContext
+
+class RuntimeServiceInterp[F[_]](blocker: Blocker,
+                                 semaphore: Semaphore[F],
+                                 config: RuntimeServiceConfig,
+                                 authProvider: LeoAuthProvider[F],
+                                 serviceAccountProvider: ServiceAccountProvider[F],
+                                 dockerDAO: DockerDAO[F],
+                                 googleStorageService: GoogleStorageService[F],
+                                 publisherQueue: fs2.concurrent.Queue[F, LeoPubsubMessage])(implicit F: Async[F],
+                                                                                            log: StructuredLogger[F],
+                                                                                            timer: Timer[F],
+                                                                                            cs: ContextShift[F],
+                                                                                            dbReference: DbReference[F],
+                                                                                            ec: ExecutionContext)
+    extends RuntimeService[F] {
+
+  def createRuntime(userInfo: UserInfo,
+                    googleProject: GoogleProject,
+                    runtimeName: RuntimeName, //TODO: rename this to RuntimeName once Rob's PR is in
+                    req: CreateRuntime2Request)(implicit as: ApplicativeAsk[F, RuntimeServiceContext]): F[Unit] =
+    for {
+      context <- as.ask
+      implicit0(traceId: ApplicativeAsk[F, TraceId]) <- F.pure(ApplicativeAsk.const[F, TraceId](context.traceId)) //TODO: is there a better way to do this?
```

Comment:
Is this needed since the implicit ev is passed to this method?

Reply:
oh, this was left over before I moved it to route

Reply:
actually, this is needed because `ApplicativeAsk[F, RuntimeServiceContext]` is different from `ApplicativeAsk[F, TraceId]`....there might be a way to derive `ApplicativeAsk[F, TraceId]` from `ApplicativeAsk[F, RuntimeServiceContext]` , but I didn't spend time look into it